c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2674,7 +2674,12 @@ class DeviceCachingAllocator {
         // any potential exceptions in the cudaMallocMaybeCapturing function.
         auto sg = c10::make_scope_exit([&]() { lock.lock(); });
         lock.unlock();
-        p.err = cudaMallocMaybeCapturing(&ptr, size);
+      }
+      auto active_pool = MemPoolContext::getActiveMemPool();
+      if (active_pool && active_pool->allocator() &&
+          p.pool->owner_PrivatePool) {
```

Comment:
> @ezyang For my education, why do we have to test owner_PrivatePool?

We only want to customize the allocations that are going into the `graph_pools` container, and the entries for that container is created in beginAllocateToPool: https://github.com/pytorch/pytorch/blob/27538671ae66ceafe897b44317fe8a4184e13b8b/c10/cuda/CUDACachingAllocator.cpp#L2051. So I just wanted to be extra defensive about it if somebody decided to not using beginAllocateToPool.

Reply:
If not using beginAllocateToPool is disallowed, then just add an ASSERT about this, don't paper over bugs haha